### PR TITLE
[WIP] Partial Reliability

### PR DESCRIPTION
**Pull request by enobufs**
Added partial reliability and testing

Relates to pions/webrtc#334

### CODE REVIEW

**Comment by enobufs** [comment]
@Sean-Der could you help on go-modules?  How can I have this branch depend on pions/sctp#sctp-pr?

**Comment by enobufs** [comment]
Why do I get this error? My local commit-msg does not complain...
```
it failed 'Separate subject from body with a blank line'
```

**Comment by Sean-Der** [comment]
Hey @enobufs 

I think that error is coming from the merge commit, if you rebase and force push it should fix that. The local hook I think only does the current commit (travs-ci tests every new commit though)

**Comment by Sean-Der** [comment]
Really nice work on all the testing! that is amazing :+1:

File: go.sum [hunk 1]
```diff
@@ -1,15 +1,24 @@
...
+github.com/davecgh/go-spew v1.1.1/go.mod h1:J7Y8YcW2NihsgmVo/mv3lAwl/skON4iLHjSsI+c5H38=
 github.com/pions/sctp v1.3.0 h1:FSoWK1yFhNHj7ZLfGw0+h0/YOXqkMR9C58lz59uiTMY=
 github.com/pions/sctp v1.3.0/go.mod h1:GZTG/xApE7wdUFEQq2Rmzgxl/+YaB/L1k8xUl1D5bmo=
+github.com/pions/sctp v1.3.1-0.20190210022111-5507c9cbd43e/go.mod h1:GZTG/xApE7wdUFEQq2Rmzgxl/+YaB/L1k8xUl1D5bmo=
+github.com/pions/sctp v1.3.1-0.20190210022111-c9af1d25209e h1:Ede+LP+8Gz4pVOWJGG4qXzRGJ906iCv5fLtMPAyICpo=
+github.com/pions/sctp v1.3.1-0.20190210022111-c9af1d25209e/go.mod h1:GZTG/xApE7wdUFEQq2Rmzgxl/+YaB/L1k8xUl1D5bmo=
+github.com/pions/sctp v1.3.2-0.20190211230616-5507c9cbd43e h1:pA7x80ZbqAnIKZzW7RS8lcx3XOQlKyf1HSkLVfC3A4g=
+github.com/pions/sctp v1.3.2-0.20190211230616-5507c9cbd43e/go.mod h1:GZTG/xApE7wdUFEQq2Rmzgxl/+YaB/L1k8xUl1D5bmo=
 github.com/pions/transport v0.0.0-20190110151433-e7cbf7d5f464 h1:/xAqYLQMIU2ci1fhEdgaFq1zc9E2p83+3a02LzUDRvQ=
 github.com/pions/transport v0.0.0-20190110151433-e7cbf7d5f464/go.mod h1:HLhzI7I0k8TyiQ99hfRZNRf84lG76eaFnZHnVy/wFnM=
```
Comment (backkem) [inline go.sum]: Maybe run `go mod tidy` to cleanup the old dependencies.

File: datachannel_test.go [hunk 2]
```diff
@@ -0,0 +1,522 @@
...
+	"time"
+
+	"github.com/pions/sctp"
+	"github.com/stretchr/testify/assert"
+)
+
+////////////////////////////////////////////////////////////////////////////////
+// testConn, connBridge for emulating dtls.Conn
+
+type testConn struct {
```
[the remark this answers is not in the document]
Comment (enobufs) [reply datachannel_test.go]: I wondered about that. Will do!

File: go.sum (hunk 1 above)
Comment (enobufs) [reply go.sum]: Thank you. (I should sit down and read go-modules document properly...)

File: datachannel_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (enobufs) [reply datachannel_test.go]: Done at e0de5

File: go.sum (hunk 1 above)
Comment (enobufs) [reply go.sum]: Done at e0de5

File: datachannel.go [hunk 3]
```diff
@@ -24,6 +24,27 @@ type Config struct {
...
+		stream.SetReliabilityParams(true, sctp.ReliabilityTypeReliable, config.ReliabilityParameter)
+	case ChannelTypePartialReliableRexmit:
+		stream.SetReliabilityParams(false, sctp.ReliabilityTypeRexmit, config.ReliabilityParameter)
+	case ChannelTypePartialReliableRexmitUnordered:
+		stream.SetReliabilityParams(true, sctp.ReliabilityTypeRexmit, config.ReliabilityParameter)
+	case ChannelTypePartialReliableTimed:
+		stream.SetReliabilityParams(false, sctp.ReliabilityTypeTimed, config.ReliabilityParameter)
+	case ChannelTypePartialReliableTimedUnordered:
+		stream.SetReliabilityParams(true, sctp.ReliabilityTypeTimed, config.ReliabilityParameter)
+	default:
```
Comment (Sean-Der) [inline datachannel.go]: This default case should throw an error (in case we get a config.ChannelType that is in a bad state)

**Pull request merged**